x-pack/plugins/apm/common/apm_saved_object_constants.ts

```diff
@@ -8,10 +8,22 @@
 // the types have to match the names of the saved object mappings
 // in /x-pack/plugins/apm/mappings.json
 
-// APM indices
+// APM indices (legacy)
+/**
+ * Should use APM_INDICES_SPACE_SAVED_OBJECT_TYPE instead
+ * @deprecated
+ * */
 export const APM_INDICES_SAVED_OBJECT_TYPE = 'apm-indices';
+/**
+ * Should use APM_INDICES_SPACE_SAVED_OBJECT_ID instead
+ * @deprecated
+ * */
 export const APM_INDICES_SAVED_OBJECT_ID = 'apm-indices';
 
+// APM indices (space aware)
+export const APM_INDICES_SPACE_SAVED_OBJECT_TYPE = 'apm-indices-space';
+export const APM_INDICES_SPACE_SAVED_OBJECT_ID = 'apm-indices-space';
```

Comment:
I agree that "apm-indices-space" is a little confusing 😅 

Minor suggestions:

* Rename the `APM_INDICES_SAVED_OBJECT_TYPE` constant to `APM_INDICES_LEGACY_SAVED_OBJECT_TYPE`, and move it out of this common file since it's not needed by other consumers anymore, it can go in `get_apm_indices.ts` instead
* Change `'apm-indices-space'` to `'apm-index-settings'`, and rename `APM_INDICES_SPACE_SAVED_OBJECT_` to `APM_INDEX_SETTINGS_SAVED_OBJECT_`